### ISSUE TITLE
Does sarama support setting a consumer group.instance.id ?

### ISSUE THREAD

**Issue by zamai** [Shopify/sarama#1885]
I'm trying to implement [Static Membership](https://www.confluent.io/blog/kafka-rebalance-protocol-static-membership/) for the consumers. By documentation config value [group.instance.id](https://docs.confluent.io/platform/current/installation/configuration/consumer-configs.html#consumerconfigs_group.instance.id) must be set on the consumer side, reading thought the docs and example I did find a way to set this value, is there a way to set it that I missed ? or is this not supported by the library?

**Comment by rodrigo-araujo-cmd** [comment]
Also would like to know that :(

**Comment by pmcmd** [comment]
+1

**Comment by totalys** [comment]
+1

**Comment by fandradesantos** [comment]
Hi, 
+1 here...
have anyone figure it out?

**Comment by qcorporation** [comment]
+1 here, anyone figured this out?

**Comment by nikolaydimitrov** [comment]
+1

**Comment by erfan-mehraban** [comment]
@dnwe

**Comment by dnwe** [comment]
Currently Sarama does _not_ support static membership ([KIP-345](https://cwiki.apache.org/confluence/display/KAFKA/KIP-345%3A+Introduce+static+membership+protocol+to+reduce+consumer+rebalances))

It wouldn't be a quick fix unfortunately as it require full updates to the various protocol types (JoinGroup, SyncGroup etc.) as most of those have not been updated since their protocol from Kafka 1.1 and earlier. Once those changes are in then we could look at exposing it as a consumer group property.

**Comment by erfan-mehraban** [comment]
@dnwe is there any road map or ETA for this feature? 

Because many new services are deployed on K8s and this feature has a great impact on their performance.

**Comment by dnwe** [comment]
I agree it would be a KIP worth implementing, but I can't really give any timeline on when it could be implemented. As I mentioned the protocol updates would be the first step and I'd really like to also move us towards using ApiVersionRequest responses for determining which protocol versions to use
